drivers/mm/mm_drv_bank.c

```diff
@@ -21,18 +21,16 @@
 
 void sys_mm_drv_bank_init(struct mem_drv_bank *bank, uint32_t bank_pages)
 {
-	bank->unmapped_pages = bank_pages;
-	bank->mapped_pages = 0;
-	bank->max_mapped_pages = 0;
+	bank->unmapped_pages = 0;
+	bank->mapped_pages = bank_pages;
+	bank->max_mapped_pages = bank_pages;
 }
 
 uint32_t sys_mm_drv_bank_page_mapped(struct mem_drv_bank *bank)
 {
 	bank->unmapped_pages--;
-	bank->mapped_pages++;
-	if (bank->mapped_pages > bank->max_mapped_pages) {
-		bank->max_mapped_pages = bank->mapped_pages;
-	}
+	if (bank->mapped_pages < bank->max_mapped_pages)
+		bank->mapped_pages++;
 	return bank->mapped_pages;
```

Comment:
Why we are always decrementing `unmapped_pages` value, but `mapped_pages` is incremented only when we didn't get to the max value?

Reply:
The values are determined by HW definitions. Adding asserts.